### DISCUSSION TITLE
How to display the URL and name of the parent page in a layout file?

### DISCUSSION THREAD

**Discussion by loveeric** [datenstrom/community#845] [Ask a question] [answered]
Hello,

How to display the URL and name of the parent page on the normal page.

I'm not a developer, but I'm trying to modify the theme myself to do what I want.

I am using "$this->yellow->page->getParent()", but it cannot be displayed correctly, and an error has occurred. Prompt Object of class YellowPage could not be converted to string.

How can I use getParent(), or other methods to display the URL and name of the parent page?

Thanks.

**Comment by JetGum** [comment]
Hi, just a push (not an answer, unfortunately since I am looking for exactly the same thing!
Cheers and thanks!

**Reply by schulle4u** [reply]
Tipp: Use the upvote button if you are interested to keep an unanswered question on top. 🙂

**Reply by JetGum** [reply]
Thanks @schulle4u for the tipp!

**Comment by PetersOtto** [comment]
I don't understand exactly what you want to do. Can one of you explain this in more detail? 

The breadcrumb extension is not a solution? (https://github.com/annaesvensson/yellow-breadcrumb)

**Reply by JetGum** [reply]
Hi,

for my part and on subpages, I want to implement one link to the parent page in the layout html (like a link back to the category page). 

I thought that '$this->yellow->page->getParent()' would work out of the box, but it doesn't, but maybe I am using it the wrong way. It just links the current page.

Using the breadcrumb extension displays links to all parent pages up until root, which is sth. that I do not want.

Thanks.

**Reply by PetersOtto** [reply]
I use Javascript for this. Maybe this could be a solution for you.

If the page is called from outside (google), then it goes to Home. Otherwise it always goes back one page.

**html (blog.html):**

`<div id="back"><a href="<?php echo $this->yellow->system->getHtml("coreServerBase") ?>/">zurück</a></div>`

**custom.js:**

```
if (document.getElementById('back')) {
  if(document.referrer.includes(window.location.hostname)){
    document.getElementById('back').innerHTML = '<a href="javascript:window.history.go(-1);"></a>';
  }
}
```

**Comment by markseuffert** [comment, marked as answer]
As you wrote `$this->yellow->page->getParent()` gives access to the parent page. Here's an example:

```
<?php $page = $this->yellow->page->getParent() ?>
<?php if ($page): ?>
<a href="<?php echo $page->getLocation(true) ?>"><?php echo $page->getHtml("titleNavigation") ?></a>
<?php endif ?>
```

**Reply by JetGum** [reply]
Thanks Mark. this is the code that works for subpages.
Cheers, Carsten
